app/main.py

```diff
@@ -1 +1,33 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self, name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return "{Name: " + self.name + ", Health: " + str(self.health) + \
+               ", Hidden: " + str(self.hidden) + "}"
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbi: Herbivore) -> None:
+        if herbi in Animal.alive:
+            if herbi.hidden is False:
+                if isinstance(herbi, Herbivore):
+                    herbi.health -= 50
+                    if herbi.health <= 0:
```

Comment:
No need to do such a nesting. You can use 1 if to reduce animal's health, and after that, another separate if to remove from alive list

Reply:
done